src/dbtPowerUserExtension.ts

```diff
@@ -85,5 +85,6 @@ export class DBTPowerUserExtension implements Disposable {
     this.dbtProjectContainer.initializeWalkthrough();
     await this.dbtProjectContainer.detectDBT();
     await this.dbtProjectContainer.initializeDBTProjects();
+    this.dbtProjectContainer.validateCredentialsOnConfigurationChange();
```

Comment:
Can you create a validators folder and create a collection object, similar to the other top level constructs?